### PR TITLE
[Lib] Add some infrastructure for using Eigen library

### PR DESCRIPTION
**Pull request by Kelvin-Ng**
Issue #197

### CODE REVIEW

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -31,6 +31,12 @@ target_link_libraries(Vector ${husky})
 target_link_libraries(Vector ${HUSKY_EXTERNAL_LIB})
 husky_default_properties(Vector)
 
+# VectorDeprecated
+add_executable(VectorDeprecated vector_deprecated.cpp)
```
Comment (Yuzhen11) [inline examples/CMakeLists.txt]: missing vector_deprecated.cpp?

Comment (Kelvin-Ng) [inline examples/CMakeLists.txt]: Yes. Added.

File: lib/vector.hpp [hunk 2]
```diff
@@ -448,27 +546,7 @@ inline SparseVector<T> operator*(T c, const SparseVector<T>& a) {
...
-    friend husky::BinStream& operator>>(husky::BinStream& stream, LabeledPoint<T, U>& b) {
-        stream >> b.x >> b.y;
-        return stream;
-    }
-};
-
 #include "vector.tpp"
 
-}  // namespace lib
+}
```
Comment (kygx-legend) [inline lib/vector.hpp]: Line 473 still needed?

File: lib/vector.hpp [hunk 3]
```diff
@@ -14,7 +14,12 @@
 
 #pragma once
 
-#include <algorithm>
+#define EIGEN_MATRIXBASE_PLUGIN "lib/eigen_matrix_base_plugin.hpp"
+
+#include <Eigen/Sparse>
```
Comment (kygx-legend) [inline lib/vector.hpp]: `#include "Eigen/Sparse"`

File: cmake/dep.cmake [hunk 4]
```diff
@@ -13,6 +13,27 @@
 # limitations under the License.
 
 
+### Eigen ###
+
+set(EIGEN_FIND_REQUIRED true)
```
Comment (kygx-legend) [inline cmake/dep.cmake]: I think we need to check the version of Eigen in CMake. Could you put a `TODO` here?

**Comment by Kelvin-Ng** [comment]
@kygx-legend Done

**Pull request merged**